src/config/util.ts

```diff
@@ -23,13 +25,16 @@ export function resolvePackage(id: string, paths: { paths: string[] }): string {
   return require.resolve(id, paths)
 }
 
-export function loadJSON(path: string): Promise<any> {
+export function loadJSON(_path: string): Promise<any> {
   debug('config')('loadJSON %s', path)
+  // Allows reading JSON with comments
+  const _JSON = path.extname(_path) === '.jsonc' ? JSON5 : JSON
```

Comment:
thinking out loud here, maybe using dynamic imports here (so json5 isn't loaded by json users) might be kind.